packages/shared/src/request/helpers/healthCheckRequest.sni.ts

```
@@ -0,0 +1,126 @@
+ */
+export async function healthCheckRequest(
+  config: IHealthCheckConfig,
+): Promise<IHealthCheckResponse> {
+  const { url, method = 'GET', timeout = 10_000, headers = {} } = config;
+
+  // Extract hostname from URL
+  const hostname = extractHostname(url);
+  if (!hostname) {
+    throw new OneKeyLocalError(
+      '[HealthCheck] Invalid URL - cannot extract hostname',
+    );
```

ESLint

[error] Expected an error object to be thrown.

(@typescript-eslint/no-throw-literal)